user-management/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+import { runInNewContext } from 'vm';
+
+const hashKey = process.env.TOKEN_SECRET ?? '';
+if (!hashKey) throw new Error('TOKEN_SECRET not found in env');
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void {
+  try {
+    const authHeader = req.get('Authorization');
+    const token = authHeader?.split('Bearer ')[1];
+    if (!authHeader || !token)
```

Comment:
This works great, but I thought I'd point out that if `authHeader` is undefined (because there is no header), then `token` will also be undefined because of the `?`. (`foo?.bar` returns `undefined` if `foo` is null or undefined)

So the check `if (!token)` works just as well here.